### ISSUE TITLE
Unexpected AVC Denial when running pip install during a docker build on Docker-in-Docker

### ISSUE THREAD

**Issue by backjo** [bottlerocket-os/bottlerocket#1569]
Apologies if this is expected behavior - I'm fairly new to working with SELinux. 

**Platform I'm running on:**
BottleRocket 1.1.0 - EKS 1.19
```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/psp: eks.privileged
  name: test-pod
spec:
  containers:
  - image: docker:dind
    name: dind
    securityContext:
      privileged: true
      seLinuxOptions:
        level: s0
        role: system_r
        type: super_t
        user: system_u
```

```
FROM ubuntu:20.04
RUN apt update
RUN apt install python3-pip -y
RUN pip install 'ansible-lint<5'
```

**What I expected to happen:**
When running a docker build of the above Dockerfile on a privileged pod using DinD (Docker in Docker), I'd expect the build to complete successfully.

**What actually happened:**
The docker build failed with permission denied errors.

build error:
```
 error: [Errno 13] Permission denied: '/usr/local/lib/python3.8/dist-packages/ansible_base-2.10.9.egg-info/SOURCES.txt'
```

AVC Denials:
```
[130009.198765] audit: type=1400 audit(1620841175.265:16780): avc:  denied  { relabelfrom } for  pid=3815087 comm="python3" name="SOURCES.txt" dev="overlay" ino=2008528 scontext=system_u:system_r:control_t:s0 tcontext=system_u:object_r:local_t:s0 tclass=file permissive=0
[130014.396610] audit: type=1400 audit(1620841180.465:16781): avc:  denied  { relabelfrom } for  pid=3815920 comm="python3" name="SOURCES.txt" dev="overlay" ino=556241 scontext=system_u:system_r:control_t:s0 tcontext=system_u:object_r:local_t:s0 tclass=file permissive=0
[130018.499288] audit: type=1400 audit(1620841184.565:16782): avc:  denied  { relabelfrom } for  pid=3816115 comm="python3" name="SOURCES.txt" dev="overlay" ino=558269 scontext=system_u:system_r:control_t:s0 tcontext=system_u:object_r:local_t:s0 tclass=file permissive=0
```

Dockerd seems to be running with control_t
```bash 
// From the admin container
ps -e
system_u:system_r:control_t:s0  3795910 ?        00:00:07 dockerd
```

**How to reproduce the problem:**
Run `docker build` of the mentioned Dockerfile in the home directory of the dind container.

**Comment by bcressey** [comment]
The issue is that `privileged: true` and seLinuxOptions are mutually exclusive. That's specifically because of [how containerd handles it](https://github.com/containerd/cri/blob/aa2d5a97c/pkg/server/container_create.go#L164), but also because traditionally privileged means "all the privileges".

That's not the case on Bottlerocket because we want to reserve some actions - modifying the API settings on disk; editing the generated files in `/etc`; relabeling the filesystem or rewriting the SELinux policy - for the break-glass, "I really know what I'm doing" opt-in use cases.

However, what you tried should still work, and it's a bug in our overall approach that it didn't.

You can work around it by defining your own, mostly-privileged permission set:
```
apiVersion: v1
kind: Pod
metadata:
  name: dind-pod
spec:
  volumes:
  - name: cgroupfs
    hostPath:
      path: /sys/fs/cgroup
  containers:
  - image: docker:dind
    name: dind
    volumeMounts:
    - mountPath: /sys/fs/cgroup
      name: cgroupfs
    securityContext:
      allowPrivilegeEscalation: true
      capabilities:
        add:
        - DAC_READ_SEARCH
        - NET_ADMIN
        - SYS_ADMIN
        - SYS_RESOURCE
      seccompProfile:
        type: Unconfined
      seLinuxOptions:
        user: system_u
        role: system_r
        type: super_t
        level: s0
```

This isn't a perfect substitute - it doesn't grant the ability to write to all devices, or add all capabilities - but it works to build your example Dockerfile.

**Comment by backjo** [comment]
Thanks for the assist @bcressey - I assumed there was at least some level of user error here :) 

I wonder if it might be useful to have some sort of F.A.Q/documentation/example around the specific Docker in Docker use case - I imagine issues like #1187 are probably similar in nature. I imagine other folks who are trying to use BottleRocket as a general purpose node on EKS and also run build infrastructure on EKS will probably have a similar use case.

**Comment by bcressey** [comment]
That's a good idea. We'd like to use our [Q&A](/bottlerocket-os/bottlerocket/discussions/categories/q-a) for content like that.

I want to keep this issue around for tracking a fix in containerd, but we can add your question and my answer there.

**Comment by bcressey** [comment]
This also breaks `podman` inside a container, which like `pip` tries to set the same extended attributes for the files it copies, leading to these denials when setting `security.selinux`.

I've confirmed that the changes in #1733 fix this for a `dind` or `podman` container with `privileged: true`, which eliminates the need for the above workaround - it won't be necessary to craft a privileged-like set of capabilities.

**Comment by bcressey** [comment]
This should be fixed as of the 1.3.0 release.
